cpp/src/io/parquet/experimental/hybrid_scan_helpers.cpp

```diff
@@ -249,40 +264,238 @@ aggregate_reader_metadata::select_payload_columns(
+                  output_column_schemas.end(),
+                  literals.begin(),
+                  std::back_inserter(bloom_filter_col_schemas),
+                  [](auto& bloom_filter_literals) { return not bloom_filter_literals.empty(); });
+
+  // No equality literals found, return empty vector
+  if (bloom_filter_col_schemas.empty()) { return {}; }
+
+  // Compute total number of input row groups
+  auto const total_row_groups = compute_total_row_groups(row_group_indices);
+
+  // Descriptors for all the chunks that make up the selected columns
+  auto const num_bloom_filter_columns = bloom_filter_col_schemas.size();
+  auto const num_chunks               = total_row_groups * num_bloom_filter_columns;
+
+  std::vector<byte_range_info> bloom_filter_bytes;
+  bloom_filter_bytes.reserve(num_chunks);
+
+  // Flag to check if we have at least one valid bloom filter offset
+  auto have_bloom_filters = false;
+
+  // For all sources
+  std::for_each(thrust::counting_iterator<size_t>(0),
+                thrust::counting_iterator(row_group_indices.size()),
+                [&](auto const src_index) {
+                  // Get all row group indices in the data source
+                  auto const& rg_indices = row_group_indices[src_index];
+                  // For all row groups
+                  std::for_each(rg_indices.cbegin(), rg_indices.cend(), [&](auto const rg_index) {
+                    // For all column chunks
+                    std::for_each(
+                      bloom_filter_col_schemas.begin(),
+                      bloom_filter_col_schemas.end(),
+                      [&](auto const schema_idx) {
+                        auto& col_meta = get_column_metadata(rg_index, src_index, schema_idx);
+                        // Get bloom filter offsets and sizes
+                        bloom_filter_bytes.emplace_back(col_meta.bloom_filter_offset.value_or(0),
+                                                        col_meta.bloom_filter_length.value_or(0));
+
+                        // Set `have_bloom_filters` if `bloom_filter_offset` is valid
+                        if (col_meta.bloom_filter_offset.has_value()) { have_bloom_filters = true; }
+                      });
+                  });
+                });
+
+  if (not have_bloom_filters) { return {}; }
+
+  return bloom_filter_bytes;
+}
+
+std::vector<byte_range_info> aggregate_reader_metadata::get_dictionary_page_bytes(
+  cudf::host_span<std::vector<cudf::size_type> const> row_group_indices,
+  host_span<data_type const> output_dtypes,
+  host_span<cudf::size_type const> output_column_schemas,
+  std::reference_wrapper<ast::expression const> filter)
+{
+  // Collect (in)equality literals for each input table column
+  auto const literals =
+    dictionary_literals_collector{filter.get(), static_cast<cudf::size_type>(output_dtypes.size())}
+      .get_literals();
+
+  auto iter = thrust::make_zip_iterator(thrust::make_tuple(
```

Comment:
`make_zip_iterator` does not need to take in a tuple. Only constructing `thrust::zip_iterator` does.